packages/grant-explorer/src/hooks/matchingEstimate.ts

```diff
@@ -0,0 +1,63 @@
+import useSWR from "swr";
+import { Address } from "viem";
+import { ChainId } from "common";
+
+export type MatchingEstimateResult = {
+  totalReceived: string;
+  contributionsCount: string;
+  sumOfSqrt: string;
+  capOverflow: string;
+  matchedWithoutCap: string;
+  matched: string;
+  difference: bigint;
+  differenceInUSD: number;
+  roundId: string;
+  chainId: number;
+  recipient: string;
+};
+
+type UseMatchingEstimatesParams = {
+  roundId: Address;
+  chainid: ChainId;
+  potentialVotes: {
+    contributor: string;
+    recipient: string;
+    amount: bigint;
+    token: string;
+  }[];
+};
+
+type JSONValue = string | number | boolean | bigint | JSONObject | JSONValue[];
+
+interface JSONObject {
+  [x: string]: JSONValue;
+}
```

Comment:
I don't think it can be done otherwise here but just FYI there's a [monster issue in the TypeScript repo](https://github.com/microsoft/TypeScript/issues/1897) about typing generic JSON.

Reply:
Yeah, not great looking but probs the best we can do rn.